pkgs/types/src/lib.zig

```diff
@@ -127,6 +128,18 @@ pub const BeamSTFProverInput = struct {
     state: BeamState,
 };
 
+// a super hacky cloning utility for ssz - replace by a better mechanis
+// should be upstreated into the ssz lib even with a proper impl
+pub fn sszClone(allocator: Allocator, comptime T: type, data: T) !T {
+    var bytes = std.ArrayList(u8).init(std.testing.allocator);
+    defer bytes.deinit();
+
+    try ssz.serialize(T, data, &bytes);
+    var cloned: T = undefined;
+    try ssz.deserialize(T, bytes.items[0..], &cloned, allocator);
+    return cloned;
+}
+
```

Comment:
ouch, you weren't lying when you said this is hacky. Could it be possible to not do the SSZ serialization and simply implement a `clone` function on `BeamState`? That could simply start being just this function, but over time would evolve to be a proper cloning function? I just can't imagine this approach being very efficient (I agree it's easy to maintain, though)

Reply:
yes thats the plan to move to a proper function, see comments